nbgrader/exchange/abc/collect.py

```diff
@@ -0,0 +1,26 @@
+from traitlets import Bool
+
+from .exchange import Exchange
+
+
+class ExchangeCollect(Exchange):
+
+    update = Bool(
+        False,
+        help="Update existing submissions with ones that have newer timestamps."
+    ).tag(config=True)
+
+    check_owner = Bool(
+        default_value=True,
+        help="Whether to cross-check the student_id with the UNIX-owner of the submitted directory."
+    ).tag(config=True)
+
+
+    def init_src(self):
+        pass
+
+    def init_dest(self):
+        pass
+
+    def copy_files(self):
+        pass
```

Comment:
Maybe it's designed as an abstract base class?

Reply:
They could probably be removed though.

Reply:
I left some original structure in as I wanted feedback on what people thought would be useful to keep. These methods are definitely not needed though, and I've removed them in my local branch.

Reply:
If we follow the logic of the file-based exchange - all common methods are implemented in the base `Exchange` class, with `raise NotImplementedError`, and then defined in the actual implementation classes (with a `pass` if they do nothing.

This would be good login to follow if we want to make this code as usable as possible for others.